### DISCUSSION TITLE
How to increase default font size

### DISCUSSION THREAD

**Discussion by alexandre1985** [stumpwm/stumpwm#1117] [Q&A]
I am using the default font of StumpWM.
I have tried some clx-truetype fonts, but I decided that it made my StumpWM too slow, so I don't want the clx-truetype fonts.

I want to keep the original font.
But with one minor adjustment, I want to increase slightly it's font size. How can I do this?

(set-font ...what?...)

Thank you.

**Comment by femnad** [comment]
What I have is `(set-font "-*-terminus-*-*-*-*-20-*-*-*-*-*-*-*")`. I think if you run `xfontsel` and only modify `pxlsz` or `ptSz`, it should give you a string which is the default font but with a different size.

**Comment by alexandre1985** [comment]
For that I must install terminus. I was avoiding any font install.

In this thread I am looking for a way to use the default font (no install of font) and to increase the default font.

In having not done that, I have installed `tamsyn-font` and set this in StumpWM config:

(set-font "-*-tamsyn-medium-*-*-*-17-*-*-*-*-*-*-*")

I am happy with this result. Still, if anyone knows how to increase the default font of StumpWM, I would be happy to try it out.

**Comment by femnad** [comment]
If you run `xfontsel` and change `pxlsz`, the first font which gets picked up can be considered the default font and you can narrow down to identify what `fndry`, `fmly`, `wght` etc. should be based on how it looks. Anyway I just did that and `-misc-fixed-medium-r-*--15-*-*-*-*-*-*-*` seems to be the default font on my system since it matches the font when `set-font` is not used.

**Comment by alexandre1985** [comment]
@femnad increasing the font to `-misc-fixed-medium-r-*-*-17-*-*-*-*-*-*-*` gives an error and all the fonts go blank.

**Comment by femnad** [comment]
Yeah, it looks weird in xfontsel too: 

![content-duckling](https://user-images.githubusercontent.com/1352288/213933123-7df8ac91-220e-4b09-8082-45c3a8667a9f.png)

18 doesn't seem much different, size-wise, but `-misc-fixed-medium-r-*--20-*-*-*-*-*-*-*` seems to work and looks like it's slightly bigger than the default font.

**Comment by alexandre1985** [comment]
In my StumpWM, -misc-fixed-medium-r-*--20-*-*-*-*-*-*-* gives an error, and defaults to the original font.

I am on Artix Linux, with a StumpWM compiled from source, been today version 2.3.0 .

**Comment by jamesmccabe** [comment]
The default bitmap font seems to be "9x15" AFAIK. You could try changing that with parameters from `xfontsel`.

Default font seems to be set in [primitives.lisp](https://github.com/stumpwm/stumpwm/blob/master/primitives.lisp#L527).

Logical font description string seems to be :
`-Misc-Fixed-Medium-R-Normal--15-140-75-75-C-90-ISO10646-1 `

`(set-font "10x20")` or
`(set-font "-misc-fixed-medium-r-*--20-*-*-*-*-*-*-*")` works OK for me. Still seems a bit small though.
